torchacc/utils/consolidate_and_reshard_ckpts.py

```diff
@@ -0,0 +1,107 @@
+
+DEFAULT_MODEL_NAME_PATTERN = "rank*-of-*-model.pth"
+DEFAULT_OPTIM_NAME_PATTERN = "rank*-of-*-optimizer.pth"
+
+
+def main():
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--ckpt_dir",
+        type=str,
+        required=True,
+        help=(
+            f"The name dir of the XLA FSDP checkpoint files to be consolidated and reshard. "
+            f"Files matching the pattern ``ckpt_dir + ckpt_name`` will be loaded."
+            f"For model, the default pattern is {DEFAULT_MODEL_NAME_PATTERN}. For optimizer,"
+            f"the default pattern is {DEFAULT_OPTIM_NAME_PATTERN}"),
+    )
+    parser.add_argument(
+        "--ckpt_name",
+        type=str,
+        default="",
+        help=(
+            f"The name pattern of the XLA FSDP checkpoint files to be consolidated and reshard. "
+            f"Files matching the pattern ``ckpt_dir + ckpt_name`` will be loaded."
+            f"For model, the default pattern is {DEFAULT_MODEL_NAME_PATTERN}. For optimizer,"
+            f"the default pattern is {DEFAULT_OPTIM_NAME_PATTERN}"),
+    )
+    parser.add_argument(
+        "--ckpt_type",
+        type=str,
+        choices=["model", "optimizer"],
+        default="model",
+        help=(
+            "The type of checkpoint to consolidate, you can choose model or optimizer. Please consolidate model fisrt, and then consolidate optimizer."
+        ),
+    )
+    parser.add_argument(
+        "--reshard_num",
+        type=int,
+        default=1,
+        help=(
+            "We now support the reshard of XLA FSDP checkpoint according to the reshard_num."
+        ),
+    )
+    parser.add_argument(
+        "--save_dir",
+        type=str,
+        default="",
+        help=(
+            f"The save dir of the output checkpoint files, the default value will be set to arg: ckpt_dir."
+            f"Files will be saved in path: ``save_dir + save_name``."
+            f"For consolidated checkpoint, the default path is: ``save_dir + model/optimizer_consolidated.pth``."
+            f"For reshard checkpoints, the default path is: ``save_dir + {DEFULT_MODEL_NAME_PATTERN}/{DEFAULT_OPTIM_NAME_PATTERN}``."
+        ),
+    )
+    parser.add_argument(
+        "--save_name",
+        type=str,
+        default="",
+        help=(
+            f"The save name pattern of the output checkpoint files, the default value is {DEFULT_MODEL_NAME_PATTERN}/{DEFAULT_OPTIM_NAME_PATTERN}."
+            f"Files will be saved in path: ``save_dir + save_name`.`"
+            f"For consolidated checkpoint, the default path is: ``save_dir + model/optimizer_consolidated.pth``"
+            f"For reshard checkpoints, the default path is: ``save_dir + {DEFULT_MODEL_NAME_PATTERN}/{DEFAULT_OPTIM_NAME_PATTERN}``."
+            f"For reshard checkpoints, please use the same name patthern as {DEFULT_MODEL_NAME_PATTERN} and {DEFAULT_OPTIM_NAME_PATTERN}."
+        ),
+    )
+
+    args = parser.parse_args()
+
+    if args.ckpt_type == "model":
```

Comment:
Is it suitable to add some judgement and set default value here? Or is it better to remove these code to the consolidate function.